app/src/ui/changes/filter-changes-logic.ts

```diff
@@ -0,0 +1,315 @@
+
+/**
+ * Compute the 'Include All' checkbox value
+ */
+export function getCheckAllValue(
+  workingDirectory: WorkingDirectoryStatus,
+  rebaseConflictState: RebaseConflictState | null,
+  filteredItems: Map<string, IChangesListItem>
+): CheckboxValue {
+  if (
+    filteredItems.size === workingDirectory.files.length &&
+    rebaseConflictState === null
+  ) {
+    return getCheckBoxValueFromIncludeAll(workingDirectory.includeAll)
+  }
+
+  const files = workingDirectory.files.filter(f => filteredItems.has(f.id))
+
+  if (files.length === 0) {
+    // the current commit will be skipped in the rebase
+    return CheckboxValue.Off
+  }
+
+  if (rebaseConflictState !== null) {
+    // untracked files will be skipped by the rebase, so we need to ensure that
+    // the "Include All" checkbox matches this state
+    const onlyUntrackedFilesFound = files.every(
+      f => f.status.kind === AppFileStatusKind.Untracked
+    )
+
+    if (onlyUntrackedFilesFound) {
+      return CheckboxValue.Off
+    }
+
+    const onlyTrackedFilesFound = files.every(
+      f => f.status.kind !== AppFileStatusKind.Untracked
+    )
+
+    // show "Mixed" if we have a mixture of tracked and untracked changes
+    return onlyTrackedFilesFound ? CheckboxValue.On : CheckboxValue.Mixed
+  }
+
+  const filteredStatus = WorkingDirectoryStatus.fromFiles(files)
+
+  return getCheckBoxValueFromIncludeAll(filteredStatus.includeAll)
+}
+
+/**
+ * Generate message when no files match filters
+ */
+export function getNoResultsMessage(
+  filterText: string,
+  filters: IFileFilterState
+): string | undefined {
+  if (
+    filterText === '' &&
+    !filters.includedChangesInCommitFilter &&
+    !filters.filterNewFiles &&
+    !filters.filterModifiedFiles &&
+    !filters.filterDeletedFiles &&
+    !filters.filterExcludedFiles
+  ) {
+    return undefined
+  }
+
+  const activeFilters: string[] = []
+
+  if (filterText) {
+    activeFilters.push(`"${filterText}"`)
+  }
+
+  if (filters.includedChangesInCommitFilter) {
+    activeFilters.push('Included in commit')
+  }
+
+  if (filters.filterExcludedFiles) {
+    activeFilters.push('Excluded from commit')
+  }
+
+  if (filters.filterNewFiles) {
+    activeFilters.push('New files')
+  }
+
+  if (filters.filterModifiedFiles) {
+    activeFilters.push('Modified files')
+  }
+
+  if (filters.filterDeletedFiles) {
+    activeFilters.push('Deleted files')
+  }
+
+  if (activeFilters.length === 0) {
+    return undefined
+  }
+
+  const filterList = activeFilters.join(', ')
+  return `Sorry, I can't find any changed files matching the following filters: ${filterList}`
+}
+
+/**
+ * Check if there are any active filters
+ */
+export function hasActiveFilters(
+  filterText: string,
+  filters: IFileFilterState
+): boolean {
+  return (
+    filterText !== '' ||
+    filters.includedChangesInCommitFilter ||
+    filters.filterNewFiles ||
+    filters.filterModifiedFiles ||
+    filters.filterDeletedFiles ||
+    filters.filterExcludedFiles
```

Comment:
```suggestion
    filterText !== '' || countActiveFilters(filters) > 0
```

Could simplify/reuse